sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.getCompletedApplications.e2e-spec.ts

```diff
@@ -0,0 +1,320 @@
+                .currentAssessment.offering.studyEndDate,
+            fullName: getUserFullName(
+              applicationWithDeclinedBySABCApplicationOfferingChange.student
+                .user,
+            ),
+            status:
+              declinedBySABCApplicationOfferingChange.applicationOfferingChangeRequestStatus,
+            dateCompleted:
+              declinedBySABCApplicationOfferingChange.assessedDate.toISOString(),
+          },
+          {
+            applicationNumber:
+              applicationWithDeclinedByStudentApplicationOfferingChange.applicationNumber,
+            studyStartDate:
+              applicationWithDeclinedByStudentApplicationOfferingChange
+                .currentAssessment.offering.studyStartDate,
+            studyEndDate:
+              applicationWithDeclinedByStudentApplicationOfferingChange
+                .currentAssessment.offering.studyEndDate,
+            fullName: getUserFullName(
+              applicationWithDeclinedByStudentApplicationOfferingChange.student
+                .user,
+            ),
+            status:
+              declinedByStudentApplicationOfferingChange.applicationOfferingChangeRequestStatus,
+            dateCompleted:
+              declinedByStudentApplicationOfferingChange.studentActionDate.toISOString(),
+          },
+          {
+            applicationNumber:
+              applicationWithApprovedApplicationOfferingChange.applicationNumber,
+            studyStartDate:
+              applicationWithApprovedApplicationOfferingChange.currentAssessment
+                .offering.studyStartDate,
+            studyEndDate:
+              applicationWithApprovedApplicationOfferingChange.currentAssessment
+                .offering.studyEndDate,
+            fullName: getUserFullName(
+              applicationWithApprovedApplicationOfferingChange.student.user,
+            ),
+            status:
+              approvedApplicationOfferingChange.applicationOfferingChangeRequestStatus,
+            dateCompleted:
+              approvedApplicationOfferingChange.assessedDate.toISOString(),
+          },
+        ],
+        count: 3,
+      });
+  });
+
+  it("Should return only matching completed application offering changes for the location when searched.", async () => {
+    // Arrange
+
+    // Student 1 has an approved application offering change request for the institution.
+    const approvedApplicationOfferingChange =
+      await saveFakeApplicationOfferingRequestChange(
+        db,
+        {
+          institutionLocation: collegeFLocation,
+        },
+        {
+          initialValues: {
+            applicationOfferingChangeRequestStatus:
+              ApplicationOfferingChangeRequestStatus.Approved,
+          },
+        },
+      );
+    // Student 2 has an approved with student application offering change request with a different institution.
+    const approvedApplicationOfferingChangeForDifferentLocation =
```

Comment:
Just trying to understand the value of adding the data for different location only for this particular test case and not in previous one .

Reply:
Not sure if I got your question correctly. If you are taking above the first TC in the same file, I already have different location logic, 
![image](https://github.com/bcgov/SIMS/assets/77353155/42b84a05-0a03-4af7-9699-13ceef9acb97). if you are asking for other test files, my intention was not to repeat the same scenarios in all TCs. let me know if you have any questions